enterprise/server/remote_execution/commandutil/commandutil.go

```diff
@@ -99,6 +104,42 @@ func Run(ctx context.Context, command *repb.Command, workDir string, stdin io.Re
 	}
 }
 
+// RunWithProcessGroupCleanup runs the given command, ensuring that child
+// processes are killed if the command times out.
+//
+// It is intended to be used with a command created via exec.Command(), not
+// exec.CommandContext(). Unlike exec.CommandContext.Run(), it kills the process
+// group when the context is done, instead of just killing the top-level
+// process. This helps ensure that orphaned processes aren't left running after
+// the command completes.
+//
+// It returns a FailedPrecondition error if cmd.SysProcAttr.Setpgid is not set
+// to true.
+func RunWithProcessGroupCleanup(ctx context.Context, cmd *exec.Cmd) error {
+	if cmd.SysProcAttr == nil || !cmd.SysProcAttr.Setpgid {
+		return status.FailedPreconditionError("process group cleanup requires SysProcAttr.Setpgid to be set")
```

Comment:
is this ever true? when?

Reply:
It's not true in our usage, but I figured it's better to check preconditions since this func won't work as documented if Setpgid is not set, and the error does not surface in any other way.

I also considered having this func force-set Setpgid=true, but I figured it's more error prone for this thing to be silently mutating the command.

For more context, I'm also using this func here: https://github.com/buildbuddy-io/buildbuddy/pull/1822/files#diff-edc095b5def6cdcdacd2ca035eb690ff8ad3c3391d2eaeadfbe0e0fbc2f70b24R151